### ISSUE TITLE
Can't  download the pdf  report

### ISSUE THREAD

**Issue by happyhq** [MobSF/Mobile-Security-Framework-MobSF#722]
## EXPLANATION OF THE ISSUE
{"pdf_error": "Cannot Generate PDF", "err_details": "wkhtmltopdf exited with non-zero code -6. error:\nThe switch --no-outline, is not support using unpatched qt, and will be ignored.QXcbConnection: Could not connect to display \n"}

## STEPS TO REPRODUCE THE ISSUE

```
add  apk
then download the  pdf report。
```

## CONSOLE OUTPUT

```
Internal Server Error: /PDF/
```

**Comment by superpoussin22** [comment]
which version of wkhtml do you install and on which platform ?

**Comment by ajinabraham** [comment]
Platform issue related to wkhtmltopdf. Please trouble shoot at wkhtmltopdf support

**Comment by iHTCboy** [comment]
### PDF Report Generation

You need to install wkhtmltopdf binary separately for generating PDF reports.
Check wkhtmltopdf downloads and Installing wkhtmltopdf wiki for more information.
In Windows, you need to add the folder that contains wkhtmltopdf binary to environment variable PATH.

https://github.com/MobSF/Mobile-Security-Framework-MobSF/wiki/1.-Documentation#pdf-report-generation

https://github.com/JazzCore/python-pdfkit/wiki/Installing-wkhtmltopdf

**Comment by Supraja817** [comment]
I have tried many ways to troubleshoot to generate the PDF report in MobSF but getting the following error:
![image](https://user-images.githubusercontent.com/91275032/134536927-97598e0a-4e77-48f5-838a-333d745fa407.png)
Also, the below error lines in command line :
[WARNING] 23/Sep/2021 20:56:35 - Dynamic Analysis related functions will not work.
Make sure a Genymotion Android VM/Android Studio Emulator is running before performing Dynamic Analysis.
[INFO] 23/Sep/2021 20:56:36 - Checking for Update.
[WARNING] 23/Sep/2021 20:56:37 - A new version of MobSF is available, Please update to 3.4.5 from master branch.
[ERROR] 23/Sep/2021 20:56:43 - Internal Server Error: /pdf/
ERROR:django.request:Internal Server Error: /pdf/

**Comment by happyhq** [comment]
If you can give more logs.
Maybe we can find the probem.







At 2021-09-23 23:27:53, "Supraja817" ***@***.***> wrote:

I have tried many ways to troubleshoot to generate the PDF report in MobSF but getting the following error:

Also, the below error lines in command line :
[WARNING] 23/Sep/2021 20:56:35 - Dynamic Analysis related functions will not work.
Make sure a Genymotion Android VM/Android Studio Emulator is running before performing Dynamic Analysis.
[INFO] 23/Sep/2021 20:56:36 - Checking for Update.
[WARNING] 23/Sep/2021 20:56:37 - A new version of MobSF is available, Please update to 3.4.5 from master branch.
[ERROR] 23/Sep/2021 20:56:43 - Internal Server Error: /pdf/
ERROR:django.request:Internal Server Error: /pdf/

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
